synapse/storage/engines/sqlite.py

```diff
@@ -12,12 +12,16 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import sqlite3
 import struct
 import threading
 
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
 from synapse.storage.engines import BaseDatabaseEngine
 
+if MYPY_CHECK_RUNNING:
+    import sqlite3
```

Comment:
I think you want to do something like this:

https://github.com/matrix-org/synapse/blob/229eb81498b0fe1da81e9b5b333a0285acde9446/synapse/handlers/presence.py#L50-L52